File: dotnet/console/Console/Scenarios/CreateDocumentInFolderScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using Lacuna.Signer.Api.FlowActions;
using Lacuna.Signer.Api.Folders;
using Lacuna.Signer.Api.Users;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    public class CreateDocumentInFolderScenario : Scenario
    {
        /**
         * This scenario shows step-by-step the submission of a document
         * into an already existing folder.
         */
        public override async Task RunAsync()
        {
            // 1. The file's bytes must be read by the application and uploaded
            var filePath = "sample.pdf";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);
            var uploadModel = await signerClient.UploadFileAsync(fileName, file, "application/pdf");

            // 2. Define the name of the document which will be visible in the application
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "Document in Folder Sample" };

            // 3. For each participant on the flow, create one instance of ParticipantUserModel.
            var participantUser = new ParticipantUserModel()
            {
                Name = "Jack Bauer",
                Email = "jack.bauer@mailinator.com",
                Identifier = "75502846369"
            };

            // 4. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow.
            //    This object is responsible for defining the personal data of the participant and the type of 
            //    action that he will peform on the flow.
            var flowActionCreateModel = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUser
            };

            // 5. You'll need to request the creation of a folder or get an existing one.
            var folderCreateRequest = new FolderCreateRequest()
            {
                Name = "Folder " + DateTime.UtcNow.ToString()
            };
            var folderInfoModel = await signerClient.CreateFolderAsync(folderCreateRequest);

            // 7. Send the document create request and provide the id of the folder where you wish to save the document.
            //    It's also possible to create a brand new folder by lefting the `FolderId` property as null and assigning 
            //    the property `NewFolderName` with the desired name for the folder.
            var documentRequest = new CreateDocumentRequest()
            {
                Files = new List<FileUploadModel>() { fileUploadModel },
                FlowActions = new List<FlowActionCreateModel>() { flowActionCreateModel },
                FolderId = folderInfoModel.Id
            };
            var result = (await signerClient.CreateDocumentAsync(documentRequest)).First();

            System.Console.WriteLine($"Document {result.DocumentId} created");
        }
    }
}

File: java/console/src/main/java/com/lacunasoftware/signer/sample/scenarios/CreateDocumentInFolderScenario.java
package com.lacunasoftware.signer.sample.scenarios;

import java.io.IOException;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import java.util.ArrayList;
import java.util.List;

import com.lacunasoftware.signer.CreateDocumentRequest;
import com.lacunasoftware.signer.FileUploadModel;
import com.lacunasoftware.signer.FlowActionCreateModel;
import com.lacunasoftware.signer.FlowActionType;
import com.lacunasoftware.signer.FolderCreateRequest;
import com.lacunasoftware.signer.FolderInfoModel;
import com.lacunasoftware.signer.ParticipantUserModel;
import com.lacunasoftware.signer.RestException;
import com.lacunasoftware.signer.UploadModel;
import com.lacunasoftware.signer.sample.Util;

public class CreateDocumentInFolderScenario extends Scenario {
    /**
    * This scenario shows step-by-step the submission of a document
    * into an already existing folder.
    */
    @Override
    public void Run() throws IOException, RestException {
        // 1. The file's bytes must be read by the application and uploaded using the method UploadFileAsync.
        byte[] content = Util.getInstance().getResourceFile("sample.pdf");
		UploadModel uploadModel = signerClient.uploadFile("sample.pdf", content, "application/pdf");

        // 2. Signer's server expects a FileUploadModel's list to create a document.
        FileUploadModel fileUploadModel = new FileUploadModel(uploadModel);
        fileUploadModel.setDisplayName("Doc in Folder " + OffsetDateTime.now(ZoneOffset.UTC).toString());
        List<FileUploadModel> fileUploadModelList = new ArrayList<>();
		fileUploadModelList.add(fileUploadModel);

        // 3. Foreach participant on the flow, you'll need to create an instance of ParticipantUserModel.
        ParticipantUserModel user = new ParticipantUserModel();
		user.setName("Jack Bauer");
		user.setEmail("jack.bauer@mailinator.com");
        user.setIdentifier("75502846369");
        
        // 4. You'll need to create a FlowActionCreateModel's instance foreach ParticipantUserModel
        //    created in the previous step. The FlowActionCreateModel is responsible for holding
        //    the personal data of the participant and the type of action that it will peform on the flow.
        FlowActionCreateModel flowActionCreateModel = new FlowActionCreateModel();
        flowActionCreateModel.setType(FlowActionType.SIGNER);
        flowActionCreateModel.setUser(user);

        // 5. Signer's server expects a FlowActionCreateModel's list to create a document.
        List<FlowActionCreateModel> flowActionCreateModelList = new ArrayList<>();
        flowActionCreateModelList.add(flowActionCreateModel);

        // 6. You'll need to request the creation of a folder or get an existing one.
        FolderCreateRequest folderCreateRequest = new FolderCreateRequest();
        folderCreateRequest.setName("Folder " + OffsetDateTime.now(ZoneOffset.UTC).toString());
        FolderInfoModel folderInfoModel = signerClient.createFolder(folderCreateRequest);

        // 7. To create the document request, use the list of FileUploadModel and the list of FlowActionCreateModel.
        //    Also, it's necessary to provide the id of the folder where you wish to save the document.
        //    It's also possible to create a brand new folder by lefting the `FolderId` property as null and assigning 
        //    the occult propertie `NewFolderName` with the desired name for the folder.
        CreateDocumentRequest documentRequest = new CreateDocumentRequest();
        documentRequest.setFiles(fileUploadModelList);
        documentRequest.setFlowActions(flowActionCreateModelList);
        documentRequest.setFolderId(folderInfoModel.getId());
        signerClient.createDocument(documentRequest);
    }
}